app/models/team.rb

```diff
@@ -0,0 +1,6 @@
+class Team
+  include Mongoid::Document
+  field :name, type: String
+  field :providers, type: Array, default: []
+  field :user_id, type: BSON::ObjectId
```

Comment:
Why is this not a has_and_belongs_to_many? It seems to replicate a lot of the functionality of that macro,